src/Spryker/Zed/EventBehavior/Business/Model/EventEntityTransferFilter.php

```diff
@@ -145,6 +149,104 @@ public function getEventTransfersAdditionalValues(array $eventTransfers, string
         return array_unique($additionalValues);
     }
 
+    /**
+     * @param \Generated\Shared\Transfer\EventEntityRequestTransfer $eventTransferRequestTransfer
+     *
+     * @return \Generated\Shared\Transfer\EventEntityResponseTransfer
+     */
+    public function getEventTransferValuesWithTimestamps(EventEntityRequestTransfer $eventTransferRequestTransfer): EventEntityResponseTransfer
+    {
+        $eventEntityResponseTransfer = new EventEntityResponseTransfer();
+        $sortedEventTransfers = $eventTransferRequestTransfer->getEventEntities()->getArrayCopy();
+
+        if (count($sortedEventTransfers) === 0) {
+            return $eventEntityResponseTransfer;
+        }
+        usort($sortedEventTransfers, fn (EventEntityTransfer $a, EventEntityTransfer $b) => $a->getTimestamp() <=> $b->getTimestamp());
+
+        $eventEntityResponseTransfer->setIds($this->getEventTransferIdsWithTimestamps($sortedEventTransfers));
+
+        if ($eventTransferRequestTransfer->getAdditionalValueName()) {
+            $eventEntityResponseTransfer->setAdditionalValues($this->getEventTransfersAdditionalValuesWithTimestamps($sortedEventTransfers, $eventTransferRequestTransfer->getAdditionalValueName()));
+        }
+
+        if ($eventTransferRequestTransfer->getForeignKeyName()) {
+            $eventEntityResponseTransfer->setForeignKeys($this->getEventTransferForeignKeysWithTimestamps($sortedEventTransfers, $eventTransferRequestTransfer->getForeignKeyName()));
+        }
+
+        return $eventEntityResponseTransfer;
+    }
+
+    /**
+     * @param array<\Generated\Shared\Transfer\EventEntityTransfer> $eventTransfers
+     *
+     * @return array<int, int>
+     */
+    protected function getEventTransferIdsWithTimestamps(array $eventTransfers): array
```

Comment:
This method is already called with the sortedEventTransfer, why sorting it again?